Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -37,121 +37,205 @@
+        nw_quic_set_initial_max_streams_bidirectional(options, 1000);
+        nw_quic_set_initial_max_streams_unidirectional(options, 1000);
+        nw_quic_set_max_datagram_frame_size(options, 65535);
+    };
 
-    // FIXME: Use nw_connection_group_set_new_connection_handler to receive incoming connections.
+    RetainPtr<nw_parameters_t> parameters = adoptNS(nw_parameters_create_webtransport_http(configureWebTransport, configureTLS, configureQUIC, NW_PARAMETERS_DEFAULT_CONFIGURATION));
+    return parameters;
 }
 
 void NetworkTransportSession::initialize(NetworkConnectionToWebProcess& connectionToWebProcess, URL&& url, CompletionHandler<void(RefPtr<NetworkTransportSession>&&)>&& completionHandler)
 {
-    // FIXME: Use nw_endpoint_create_url to support all URLs on systems where rdar://135036170 is fixed.
-    auto port = url.port() ? url.port() : defaultPortForProtocol(url.protocol());
-    if (!port)
-        return completionHandler(nullptr);
-
-    RetainPtr endpoint = adoptNS(nw_endpoint_create_host(url.host().utf8().data(), makeString(*port).utf8().data()));
+    RetainPtr<nw_endpoint_t> endpoint = adoptNS(nw_endpoint_create_url(url.string().utf8().data()));
     if (!endpoint) {
         ASSERT_NOT_REACHED();
         return completionHandler(nullptr);
     }
 
-    RetainPtr descriptor = adoptNS(nw_group_descriptor_create_multiplex(endpoint.get()));
-    if (!descriptor) {
+    RetainPtr<nw_parameters_t> parameters = createDatagramParameters();
+    if (!parameters) {
         ASSERT_NOT_REACHED();
         return completionHandler(nullptr);
     }
 
-    auto configureTLS = [](nw_protocol_options_t options) {
-        RetainPtr securityOptions = adoptNS(nw_quic_connection_copy_sec_protocol_options(options));
-        sec_protocol_options_set_peer_authentication_required(securityOptions.get(), true);
-        sec_protocol_options_set_verify_block(securityOptions.get(), makeBlockPtr([](sec_protocol_metadata_t metadata, sec_trust_t trust, sec_protocol_verify_complete_t completion) {
-            // FIXME: Hook this up with WKNavigationDelegate.didReceiveChallenge.
-            completion(true);
-        }).get(), dispatch_get_main_queue());
-        // FIXME: Pipe client cert auth into this too, probably.
-        sec_protocol_options_add_tls_application_protocol(securityOptions.get(), "h3");
-    };
-
-    RetainPtr parameters = adoptNS(nw_parameters_create_quic_stream(NW_PARAMETERS_DEFAULT_CONFIGURATION, configureTLS));
-
-    if (!parameters) {
+    RetainPtr<nw_connection_t> connection = adoptNS(nw_connection_create(endpoint.get(), parameters.get()));
+    if (!connection) {
         ASSERT_NOT_REACHED();
         return completionHandler(nullptr);
     }
 
-    RetainPtr connectionGroup = adoptNS(nw_connection_group_create(descriptor.get(), parameters.get()));
-    if (!connectionGroup) {
+    RetainPtr<nw_listener_t> listener = adoptNS(nw_listener_create_with_connection(connection.get(), parameters.get()));
```

Comment:
If we want to use nw_listener_create_with_connection it needs to be forward declared in the SPI header.  Can we use API instead?

Reply:
Updated to connection group.